### PR TITLE
fix: remove authenticator class when registed

### PR DESCRIPTION
**Pull request by olizilla**
only include the `w3ui-authenticator` class in the dom when we want to show the log in box.

otherwise the layout we choose for the register form is inherited by the logged in app layout which is undesirable

fixes https://github.com/web3-storage/w3ui/issues/351

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>

### CODE REVIEW

File: packages/react/src/Authenticator.tsx [hunk 1]
```diff
@@ -48,7 +52,7 @@ interface AuthenticatorProps {
 
 export function Authenticator ({ children, className = '' }: AuthenticatorProps): JSX.Element {
   return (
-    <AuthCore as='div' className={`w3ui-authenticator ${className}`}>
+    <AuthCore as='div' className={className}>
```
Comment (jchris) [inline packages/react/src/Authenticator.tsx:55]: So `w3ui-authenticator` means "needs authentication"? Maybe we want to rename to make clearer. Not a blocker for this PR.

**Review by jchris — APPROVED** [review]
runs correct for me on w3console

**Comment by jchris** [comment]
Is there another test flow I should be trying out?

**Review by travis — APPROVED** [review]

File: packages/react/src/Authenticator.tsx (hunk 1 above)
Comment (olizilla) [reply packages/react/src/Authenticator.tsx:55]: kinda yes. From the pattern of our components it means "the root element for this widget", and this change makes it so the authenticator doesn't affect the dom if the user has > 0 registered spaces, so it implicitly takes on that meaning. 

I'm not making bigger changes here as we are all in favour of doing less of the "you must give an email to play", but if we revisit this component we'd probably add explicit css classes for "registered" and "not-registered" or similar.

**Pull request merged**